### ISSUE TITLE
Profile HeatMap not updating correctly

### ISSUE THREAD

**Issue by lourdesr** [freeCodeCamp/freeCodeCamp#17822]
I don't know if this is an issue or not. This is the first time I'm posting here. The heat map in our public portfolio does not update to the current day/month. 

#### Steps to Reproduce

Make sure to set your profile heatmap to public.

1. Go to https://www.freecodecamp.org/yourusername
2. Scroll down to view the heat map. 

#### Browser and operating system

* Browser Name: Google Chrome
* Browser Version: Version 67.0.3396.87 (Official Build) (64-bit)
* Operating System: macOS High Sierra - version 10.13.1

#### Screenshot

![screen shot 2018-07-04 at 8 27 52 pm](https://user-images.githubusercontent.com/3199404/42296168-d73639ae-7fc8-11e8-8fce-f9314732aafc.png)

Right now, it is the month of July, and I have been working on the lessons for the past 2-3 weeks, it does not appear in the heatmap. 

If this is an actual issue, I'm willing to take this up. I'm still a beginner, but I am sort of confident and willing to give this a try, if that's okay.

Many Thanks,
Lourdes

**Comment by driverpizza** [comment]
I am also having this issue, however my heat map has never worked since I began two weeks ago...

**Comment by NotSoV** [comment]
Same issue. I'll be sure to post back if I find anything.

**Comment by Snowcat8** [comment]
same issue-today

**Comment by csuszka** [comment]
Same issue (using Chrome ver. 63.0.3239.132)

**Comment by paulbyars** [comment]
Same issue.  Streak count is accurate, but not showing in the heat map.

**Comment by shaminzn** [comment]
> Same issue. Streak count is accurate, but not showing in the heat map.
Same issue for me too with my heat map...

**Comment by mariatheresahqs** [comment]
Same issue with my "Heat Map", and my "Timeline" sometimes does not appear on the profile page. 

**Browser and operating system**
Browser Name: Google Chrome
Browser Version: Version 70.0.3538.102 (Official Build) (64-bit)
Operating System: macOS Mojave - version 10.14.1

**Comment by drecali** [comment]
Does anyone know which timezone is used by the streak counter and the heatmap? I just lost a 74 day streak even though I finished a challenge by 15:45PM KST (UTC +0900). The streak shows only 1 day but today's square on the heatmap is colored green, and on the timeline the challenge is recorded as being completed today as well. It seems like the streak counter is using one timezone, while the heatmap and timeline are using a second timezone. Can someone please confirm? It's really demoralizing as I was taking pride in watching the number grow every day and being able to share it with friends, family, and potential employers.

I understand that it may not be a high priority to fix, so at least knowing the rules of the streak would really help me understand and adjust my coding accordingly.

Screenshots: https://imgur.com/a/c47JNv9

Quoted from https://github.com/freeCodeCamp/freeCodeCamp/issues/7925#issuecomment-361716788
> I observed how challenges are recorded during the last week and basically saw challenge dates and streak are going by UTC and the heatmap is using EST...

**Comment by moT01** [comment]
Mine is getting cut off. Today is December 11. It didn't get cut off like this yesterday, or ever in the past that I've noticed.
<img width="710" alt="screen shot 2018-12-11 at 7 56 40 am" src="https://user-images.githubusercontent.com/20648924/49806475-7a262800-fd1d-11e8-9901-151990d21ec2.png">

**Comment by shaminzn** [comment]
> Mine is getting cut off. Today is December 11. It didn't get cut off like this yesterday, or ever in the past that I've noticed.
> <img alt="screen shot 2018-12-11 at 7 56 40 am" width="710" src="https://user-images.githubusercontent.com/20648924/49806475-7a262800-fd1d-11e8-9901-151990d21ec2.png">

At least something is showing up on your heat map, It's been six months since I've seen anything on my heat map... kinda gets discouraging when you can't see your progress at all...

**Comment by icatat** [comment]
Is anyone working on this issue? I would like to try to tackle it

**Comment by ayushshah2120** [comment]
I have no heat on the heat map.
And I Started From June 16th 2018.
My Recent Was On 30th January 2019 but no heat on entire Heat Map.

![image](https://user-images.githubusercontent.com/40225091/52160254-ec8d0600-26d7-11e9-9eac-07fd3c779ffc.png)

**Comment by ayushshah2120** [comment]
No Heat On My Profile's Heat Map
https://www.freecodecamp.org/shahayush
I Solved 21Challenges on 30th of January 2019 but there is no heat on heat map.

![image](https://user-images.githubusercontent.com/40225091/52725585-88bbe480-2fd7-11e9-95c1-5ac78c58f1f8.png)

**Comment by RandellDawson** [comment]
@raisedadead Any thoughts on this?

**Comment by ayushshah2120** [comment]
> @raisedadead Any thoughts on this?

any progress?

**Comment by RandellDawson** [comment]
@ayushshah2120 Currently we are in the middle of testing a beta site which will hopefully moved to production in the near future, so it could be a while before we have time to work on this issue.

**Comment by ayushshah2120** [comment]
> @ayushshah2120 Currently we are in the middle of testing a beta site which will hopefully moved to production in the near future, so it could be a while before we have time to work on this issue.

No Problem Sir. Thank You!

**Comment by raisedadead** [comment]
Hi everyone! 

Sorry about the delay in getting back on this. Thanks for your patience. 

We have had some work go in into this issue a few months back. So this issue is probably fixed or behaving better on the beta, it needs to be tested there.

That does mean that, @RandellDawson you are right, it's going to be a while before we can land the fix in production.

The fastest way to test this would be to copy the data of an affected user from the production (with their permission) and seed it over on the beta site and see the newer heatmap's behaviour.

If it all looks good we could add that as tests.

**Action items for anyone interested in overseeing this issue is:**
-  Find a user who has this issue on the production and seek their permission to use their data on the beta.
- Once permitted, notify me (@raisedadead) or @bouncey to copy over the user's data.
- Confirm the fix, or add more changes as per further investigation.

We definitely want to resolve this issue, which has been bugging us for a really long time now.

**Comment by ayushshah2120** [comment]
> Hi everyone!
> 
> Sorry about the delay in getting back on this. Thanks for your patience.
> 
> We have had some work go in into this issue a few months back. So this issue is probably fixed or behaving better on the beta, it needs to be tested there.
> 
> That does mean that, @RandellDawson you are right, it's going to be a while before we can land the fix in production.
> 
> The fastest way to test this would be to copy the data of an affected user from the production (with their permission) and seed it over on the beta site and see the newer heatmap's behaviour.
> 
> If it all looks good we could add that as tests.
> 
> **Action items for anyone interested in overseeing this issue is:**
> 
> * Find a user who has this issue on the production and seek their permission to use their data on the beta.
> * Once permitted, notify me (@raisedadead) or @Bouncey to copy over the user's data.
> * Confirm the fix, or add more changes as per further investigation.
> 
> We definitely want to resolve this issue, which has been bugging us for a really long time now.

It's Okay If You Use my Profile or My Data to work on your issue.

**Comment by RandellDawson** [comment]
@raisedadead, @Bouncey You have a user now.

**Comment by madmedmod** [comment]
is this case is still on going? It happened to me as well but might be with different cause.

I reset my account progress then when restarting there's no heat map progress, though just like others the strike count is working normally.

thank you guys who's working on this, appreciate it very much!

**Comment by ayushshah2120** [comment]
> is this case is still on going? It happened to me as well but might be with different cause.
> 
> I reset my account progress then when restarting there's no heat map progress, though just like others the strike count is working normally.
> 
> thank you guys who's working on this, appreciate it very much!

Actually, I also reset my account a while back and there's no heat map progress and strike counting is normal

**Comment by ayush-shah** [comment]
> This should be resolved on master I think.

what do i do to solve the heatmap with no progress(past and current)?

**Comment by MalcolmAnderson** [comment]
I just started getting this yesterday (5 November 2019).  My contribution heat map does not reflect my checkins for today.  

Here's my heat map showing no check-ins for today
![image](https://user-images.githubusercontent.com/1200043/68346139-344a1f00-00a8-11ea-893b-9c1a8e705ab8.png)

Here's my file where I modified a simple comment just to check heatmap functionality - checked in minutes earlier.
![image](https://user-images.githubusercontent.com/1200043/68346205-5fcd0980-00a8-11ea-8b62-c4a98f057aba.png)
